modules/cudf/src/data_frame.ts

```diff
@@ -92,10 +92,11 @@ export class DataFrame<T extends TypeMap = any> {
   /**
    * Return a new DataFrame with new columns added.
    *
-   * @param data mapping of names to new columns to add.
+   * @param data mapping of names to new columns to add or a GPU DataFrame object
    */
-  assign<R extends TypeMap>(data: SeriesMap<R>) {
-    return new DataFrame(this._accessor.addColumns(_seriesToColumns(data)));
+  assign<R extends TypeMap>(data: SeriesMap<R>|DataFrame<R>) {
+    const columns = (data instanceof DataFrame) ? data._accessor : _seriesToColumns(data);
+    return new DataFrame(this._accessor.addColumns(columns));
```

Comment:
I *think* the docs might turn out a bit nicer if the cases are explicitly split:
```
/**
 *  @param data mapping of names to new columns to add
 */
assign<R extends TypeMap>(data: SeriesMap<R>)

/**
 *  @param data a GPU DataFrame object
 */
assign<R extends TypeMap>(data: DataFrame<R>)

assign<R extends TypeMap>(data: SeriesMap<R>|DataFrame<R>) { ... }
```
but it's probably worth comparing to rendered output to be sure.

Reply:
@bryevdv the return type in this case is same (`DataFrame<T>`).

The above works with no issues if I do this:

```typescript
/**
 *  @param data mapping of names to new columns to add
 */
assign<R extends TypeMap>(data: SeriesMap<R>): DataFrame<T>

/**
 *  @param data a GPU DataFrame object
 */
assign<R extends TypeMap>(data: DataFrame<R>): DataFrame<T>

assign<R extends TypeMap>(data: SeriesMap<R>|DataFrame<R>) { ... }
```

Reply:
Although the above gets skipped in the rendered output(data_frame.js) file, if thats what you meant. Its only the single function definition

Reply:
docs seem like this: 
![image](https://user-images.githubusercontent.com/20476096/114475640-7f296980-9bad-11eb-9e70-a66c659bd133.png)

Reply:
FWIW I prefer that style of docs output, personally. But I don't think we've established any actual conventions

Reply:
+1

Reply:
pushed the changes, thanks